src/components/BarChart.tsx

```diff
@@ -11,9 +11,14 @@ const BarChart: React.FC<ChartPropsData> = ({
   const svgRef = React.useRef<SVGSVGElement | null>(null);
 
   // needed scaling info for d3
-  const width = 600;
+  let width;
   const height = 385;
   const margin = { top: 20, right: 10, bottom: 20, left: 50 };
+  if (window.innerWidth <= 600) {
```

Comment:
nice solve here for determining width mathematically

Reply:
thanks that was the only solution I could find at the moment.